### DISCUSSION TITLE
run chezmoi add whenever a managed file is updated on disk

### DISCUSSION THREAD

**Discussion by zany130** [twpayne/chezmoi#3686] [Q&A]
I am pretty new to dot file managers but I like how chezmoi doesn't (at least by default I know it does have the option) symlink your dotfiles into a central location. I prefer NOT to do anything special like that with them.

However one downside is I have to be constantly running `chezmoi add`. I saw this https://www.chezmoi.io/user-guide/advanced/use-chezmoi-with-watchman/ and subsequently this https://github.com/twpayne/chezmoi/issues/3680

but I haven't been successful in adjusting this for ALL files managed by chezmoi

this is what I have so far

```bash
#!/bin/bash

WATCHED_DIR="$(chezmoi managed)"

while inotifywait -e create -e modify -attrib "$WATCHED_DIR"; do
    chezmoi add "$WATCHED_DIR"
done

echo "inotifywait encountered an error, exiting the script"
exit 1
```
and this gives

```
line 5: /usr/bin/inotifywait: Argument list too long
```

the problem seems to be `chezmoi managed` lists to many files and overloads bash  see https://askubuntu.com/a/1049726

anyone have any ideas? I really would rather not use symlinks...

EDIT: probably would help to post an inxi....

```
inxi -b
System:
  Host: Garuda-Linux Kernel: 6.8.4-1-cachyos arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.0.3 Distro: Garuda Linux
Machine:
  Type: Desktop Mobo: ASRock model: X470 Taichi serial: <superuser required>
    UEFI: American Megatrends v: P5.10 date: 10/20/2022
CPU:
  Info: 6-core AMD Ryzen 5 5600X [MT MCP] speed (MHz): avg: 3809
    min/max: 550/4687
Graphics:
  Device-1: AMD Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M/6850M XT]
    driver: amdgpu v: kernel
  Display: wayland server: X.org v: 1.21.1.12 with: Xwayland v: 23.2.5
    compositors: 1: Gamescope 2: kwin_wayland driver: X: loaded: amdgpu
    unloaded: modesetting,radeon dri: radeonsi gpu: amdgpu resolution:
    1: 2048x864 2: 1396x785 3: 1536x864
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.0.4-arch1.2
    renderer: AMD Radeon RX 6700 XT (radeonsi navi22 LLVM 17.0.6 DRM 3.57
    6.8.4-1-cachyos)
Network:
  Device-1: Intel Dual Band Wireless-AC 3168NGW [Stone Peak] driver: iwlwifi
  Device-2: Intel I211 Gigabit Network driver: igb
Drives:
  Local Storage: total: 3.64 TiB used: 4.05 TiB (111.4%)
Info:
  Memory: total: 32 GiB available: 31.26 GiB used: 20.73 GiB (66.3%)
  Processes: 534 Uptime: 8h 2m Shell: fish inxi: 3.3.33
  ```

**Comment by twpayne** [comment]
`inotifywait` has a `--fromfile` option:

> `--fromfile <file>`
> Read filenames to watch or exclude from a file, one filename per line. If filenames begin with @ they are excluded as described above. If <file> is '-', filenames are read from standard input. Use this option if you need to watch too many files to pass in as command line arguments.

**Reply by zany130** [reply]
thanks   just saw that to and tried it but it fails with 

```
inotifywait -e create -e modify -e attrib -r --fromfile chezmoi-managed
double free or corruption (out)
fish: Job 1, 'inotifywait -e create -e modify…' terminated by signal SIGABRT (Abort)
```
I got the file `chezmoi-managed` by doing this
`chezmoi managed  >> chezmoi-managed`

not to sure but that sounds like maybe a bug in notify?

just a thought out loud but I wonder if its because some of the paths have spaces...

EDIT: something I also just thought about `chezmoi-manged` prints the files relative to the home directory. in this case it probably doesn't matter since I am running this from the top level of my home folder to test, but probably something to consider when I try to get it to run from the script...

**Reply by twpayne** [reply]
> not to sure but that sounds like maybe a bug in notify?

Yes, that error message comes from `inotifywait`.

> EDIT: something I also just thought about `chezmoi-manged` prints the files relative to the home directory.

`chezmoi managed` has a [`--path-style` option](https://www.chezmoi.io/reference/commands/managed/#-p-path-style-absoluterelativesource-absolutesource-relative) that allows you to change the path format, if you want.
